doc/search-metrics.org

```diff
@@ -82,11 +82,12 @@ GET /ctia/incident/topn?&aggregate-on=status&limit=3&from=2020-01-01
 #+END_SRC
 it will return for instance
 #+BEGIN_SRC javascript
-{
- "Closed": 12,
- "Open": 6,
- "New": 2
-}
+{"status": {"Closed": 12,
+            "Open": 6,
+            "New": 2},
+ "type": "topn",
+ "from": "2020-01-01",
+ "to": "2020-05-01"}
```

Comment:
You merge the result with the query params. Not easy to extract the result from the whole response. Wouldn't it be better to separate the query params in a submap with all metas of the request.

Reply:
There is only `type`, `from` and `to` that are at the route, like the aggregated field. Then all the optional search filters are put in a `filters` field:
https://github.com/threatgrid/ctia/pull/897/files#diff-e59b44685eb63cb4899cbef38c1d7d94R159

Reply:
Why `from` and `to` are not in the filters field?

Reply:
well, even if they act as a filter to they are also some kind of metadata for the computed metric. Unlike other filters they are somehow dynamic while filters are exactly the user input.
`to` is optional and set to the query date.
`from` is mandatory but coerced to `to` - 1 year if the interval exceed 1 year.
I prefer this version but I can put it in `filters` if you consider that it would be more consistent.

Reply:
I agree that I would have naturally but all the metas info in a submap and also put the result of the aggregations inside a `:data` because it appear the field is not always `"status"` but could be anything. What if I want to aggregate a field named "from" or `type` (in observable I think) or `filters` or if we want to add more metas in the future? Isn't there a collision risk?

Wouldn't it be safer to envelop the result in something like this?

```
{:data {"status": ...}
 :metas {:from ... :to ... :type ... :filters ...}}
```

Reply:
Considering this last argument and after a sync with @yogsototh I propose the following schema:
```clojure
(s/defschema AggResult
  {:data {:keyword s/Any}
   :type (s/enum :topn :histogram :cardinality)
   :filters (st/open-schema
                {:from s/Inst
                 :to s/Inst}}
```

Reply:
as some of my remarks during my review, I think you should not use `:keyword` for the keys of `:data` because it prevent you to support search on field containing special chars. I would use `string` here.

Reply:
Sorry, my comment contains a typo:
```
(s/defschema AggResult
  {:data {s/Keyword s/Any}
   :type (s/enum :topn :histogram :cardinality)
   :filters (st/open-schema
                {:from s/Inst
                 :to s/Inst}}
```
However, I do not understand your comment, that's the schema of the response here.